### PR TITLE
[Serve] Make `HANDLE_METRIC_PUSH_INTERVAL_S` configurable by env var

### PR DESCRIPTION
**Pull request by bewestphal**
## What changed?
Makes the ray serve private constant `HANDLE_METRIC_PUSH_INTERVAL_S` configurable with environment variable
`RAY_SERVE_HANDLE_METRIC_PUSH_INTERVAL_S`.

## Why are these changes needed?

Running serve 2.3.0 with Kuberay v0.4.0 when autoscaling config was enabled for an endpoint, the number of `record_handle_metrics` tasks created grew out of control, resulting in server crashes.

The `HANDLE_METRIC_PUSH_INTERVAL_S` is only necessary to autoscale endpoints from zero replicas up. So short of debugging why the tasks grow, configuring this environment variable to an unobtainable number will allow bypassing the `record_handle_metrics` tasks from being created when the minimum number of replicas desired is greater than zero.

See slack thread for debugging information:
https://ray-distributed.slack.com/archives/CNCKBBRJL/p1677188591700029

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sihanwang41 — APPROVED** [review]
Thank you for this change!

File: python/ray/serve/_private/constants.py [hunk 1]
```diff
@@ -103,7 +103,9 @@
 CLIENT_POLLING_INTERVAL_S: float = 1
 
 # Handle metric push interval. (This interval will affect the cold start time period)
-HANDLE_METRIC_PUSH_INTERVAL_S = 10
+HANDLE_METRIC_PUSH_INTERVAL_S = int(
```
Comment (shrekris-anyscale) [inline python/ray/serve/_private/constants.py:106]:
```suggestion
HANDLE_METRIC_PUSH_INTERVAL_S = float(
```

Comment (shrekris-anyscale) [reply python/ray/serve/_private/constants.py]: Awesome, thanks!

**Review by shrekris-anyscale — APPROVED** [review]
Thanks for the contribution!

**Review by GeneDer — APPROVED** [review]
I feel if there is a need for changing this, probably better to include it in the autoscaling config, but good as first step!

**Pull request merged**